clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -104,6 +109,9 @@ public void handleMessage(EntityResponse messageFromServer) {
 
       @Override
       public byte[] createExtendedReconnectData() {
+        for (ReconnectListener reconnectListener : reconnectListeners) {
```

Comment:
If it doesn't matter to miss a listener because it will be processed later, it's fine. Otherwise, it should synchronize instead.